File: 3289_The_Two_Sneaky_Numbers_of_Digitville/solution.py
# class Solution:
#     def getSneakyNumbers(self, nums: List[int]) -> List[int]:
#         freq = {}
#         result = []
#         for num in nums:
#             freq[num] = freq.get(num, 0) + 1
#         for key, value in freq.items():
#             if value == 2:
#                 result.append(key)
#         return result

class Solution:
    def getSneakyNumbers(self, nums: List[int]) -> List[int]:
        seen, result = set(), []
        for num in nums:
            if num in seen:
                result.append(num)
            else:
                seen.add(num)
        return result

File: 3289_The_Two_Sneaky_Numbers_of_Digitville/solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* getSneakyNumbers(int* nums, int numsSize, int* returnSize) {
    int* result = (int*)malloc(2 * sizeof(int));
    *returnSize = 0;

    int seen[100] = {0};

    for (int i = 0; i < numsSize; i++) {
        if (seen[nums[i]]) {
            result[*returnSize] = nums[i];
            (*returnSize)++;
        } else {
            seen[nums[i]] = 1;
        }
    }

    return result;
}